page_guard_manager.cpp:113:28: error: size of array ‘s_alt_stack’ is not an integral constant-expression   113 | static uint8_t s_alt_stack[SIGSTKSZ];

```
$ mkdir build && cd build/ && cmake ../
-- The C compiler identification is GNU 11.1.1
-- The CXX compiler identification is GNU 11.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib64/ccache/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib64/ccache/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Vulkan Header = /home/mikhail/packaging-work/git/gfxreconstruct/external/Vulkan-Headers/include/vulkan/vulkan_core.h
-- Detected Vulkan Version 1.2.176
-- Found LZ4: /usr/lib64/liblz4.so  
-- Found ZSTD: /usr/lib64/libzstd.so  
-- Found XCB: /usr/lib64/libxcb.so;/usr/lib64/libxcb-keysyms.so;/usr/lib64/libX11-xcb.so  
-- Found WAYLAND: /usr/lib64/libwayland-client.so  
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.11") 
-- Found X11: /usr/include   
-- Looking for XOpenDisplay in /usr/lib64/libX11.so
-- Looking for XOpenDisplay in /usr/lib64/libX11.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for clock_gettime in c
-- Looking for clock_gettime in c - found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/mikhail/packaging-work/git/gfxreconstruct/build

$ make
[  1%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/argument_parser.cpp.o
[  2%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/file_output_stream.cpp.o
[  3%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/file_path.cpp.o
[  4%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/hash.cpp.o
[  5%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/image_writer.cpp.o
[  6%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/keyboard.cpp.o
[  7%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/logging.cpp.o
[  8%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/lz4_compressor.cpp.o
[  9%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/zlib_compressor.cpp.o
[ 10%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/zstd_compressor.cpp.o
[ 11%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/memory_output_stream.cpp.o
[ 12%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/monotonic_allocator.cpp.o
[ 13%] Building CXX object framework/util/CMakeFiles/gfxrecon_util.dir/page_guard_manager.cpp.o
In file included from /usr/include/signal.h:328,
                 from /home/mikhail/packaging-work/git/gfxreconstruct/framework/util/platform.h:52,
                 from /home/mikhail/packaging-work/git/gfxreconstruct/framework/util/logging.h:28,
                 from /home/mikhail/packaging-work/git/gfxreconstruct/framework/util/page_guard_manager.cpp:27:
/home/mikhail/packaging-work/git/gfxreconstruct/framework/util/page_guard_manager.cpp:113:28: error: size of array ‘s_alt_stack’ is not an integral constant-expression
  113 | static uint8_t s_alt_stack[SIGSTKSZ];
      |                            ^~~~~~~~
make[2]: *** [framework/util/CMakeFiles/gfxrecon_util.dir/build.make:244: framework/util/CMakeFiles/gfxrecon_util.dir/page_guard_manager.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:384: framework/util/CMakeFiles/gfxrecon_util.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

After running that command, then compiling vulkansdk gfxreconstruct 

In file included from /usr/include/signal.h:328,
                 from /home/steve/vulkan/1.2.198.1/source/gfxreconstruct/framework/util/platform.h:52,
                 from /home/steve/vulkan/1.2.198.1/source/gfxreconstruct/framework/util/logging.h:28,
                 from /home/steve/vulkan/1.2.198.1/source/gfxreconstruct/framework/util/page_guard_manager.cpp:27:
/home/steve/vulkan/1.2.198.1/source/gfxreconstruct/framework/util/page_guard_manager.cpp:113:28: error: size of array ‘s_alt_stack’ is not an integral constant-expression
  113 | static uint8_t s_alt_stack[SIGSTKSZ];
      |                            ^~~~~~~~

--- a/framework/util/page_guard_manager.cpp
+++ b/framework/util/page_guard_manager.cpp
@@ -110,7 +110,8 @@ const uint32_t kGuardNoProtect        = PROT_READ | PROT_WRITE;
 static struct sigaction s_old_sigaction = {};
 static stack_t          s_old_stack     = {};
 
-static uint8_t s_alt_stack[SIGSTKSZ];
+static uint8_t* s_alt_stack      = nullptr;
+static size_t   s_alt_stack_size = 0;
 
 static void PageGuardExceptionHandler(int id, siginfo_t* info, void* data)
 {
@@ -188,11 +189,24 @@ static void PageGuardExceptionHandler(int id, siginfo_t* info, void* data)
 
 PageGuardManager* PageGuardManager::instance_ = nullptr;
 
+void PageGuardManager::InitializeSystemExceptionContext(void)
+{
+#if defined(__linux__)
+    if (s_alt_stack == nullptr)
+    {
+        s_alt_stack_size = SIGSTKSZ;
+        s_alt_stack      = new uint8_t[s_alt_stack_size];
+    }
+#endif
+}
+
 PageGuardManager::PageGuardManager() :
     exception_handler_(nullptr), exception_handler_count_(0), system_page_size_(GetSystemPageSize()),
     system_page_pot_shift_(GetSystemPagePotShift()), enable_copy_on_map_(kDefaultEnableCopyOnMap),
     enable_separate_read_(kDefaultEnableSeparateRead), enable_read_write_same_page_(kDefaultEnableReadWriteSamePage)
-{}
+{
+    InitializeSystemExceptionContext();
+}
 
 PageGuardManager::PageGuardManager(bool enable_copy_on_map,
                                    bool enable_separate_read,
@@ -201,7 +215,9 @@ PageGuardManager::PageGuardManager(bool enable_copy_on_map,
     exception_handler_count_(0), system_page_size_(GetSystemPageSize()),
     system_page_pot_shift_(GetSystemPagePotShift()), enable_copy_on_map_(enable_copy_on_map),
     enable_separate_read_(enable_separate_read), enable_read_write_same_page_(expect_read_write_same_page)
-{}
+{
+    InitializeSystemExceptionContext();
+}
 
 PageGuardManager::~PageGuardManager()
 {
@@ -358,7 +374,7 @@ void PageGuardManager::AddExceptionHandler()
                 stack_t new_stack;
                 new_stack.ss_sp    = s_alt_stack;
                 new_stack.ss_flags = 0;
-                new_stack.ss_size  = sizeof(s_alt_stack);
+                new_stack.ss_size  = s_alt_stack_size;
 
                 sigaltstack(&new_stack, &s_old_stack);
 

--- a/framework/util/page_guard_manager.h
+++ b/framework/util/page_guard_manager.h
@@ -181,7 +181,6 @@ class PageGuardManager
   private:
     size_t GetSystemPageSize() const;
     size_t GetSystemPagePotShift() const;
-    size_t InitSystemExceptionHandler() const;
     void   InitializeSystemExceptionContext();
 
     void AddExceptionHandler();

--- a/framework/util/page_guard_manager.h
+++ b/framework/util/page_guard_manager.h
@@ -181,6 +181,8 @@ class PageGuardManager
   private:
     size_t GetSystemPageSize() const;
     size_t GetSystemPagePotShift() const;
+    size_t InitSystemExceptionHandler() const;
+    void   InitializeSystemExceptionContext();
 
     void AddExceptionHandler();
     void RemoveExceptionHandler();

